Source/WebCore/platform/graphics/ImageSource.cpp

```diff
@@ -26,705 +26,23 @@
-        return;
-
-    ASSERT(m_decodedSize >= decodedSize);
-    m_decodedSize -= decodedSize;
-    decodedSizeChanged(-static_cast<long long>(decodedSize));
-}
-
-void ImageSource::decodedSizeReset(unsigned decodedSize)
-{
-    ASSERT(m_decodedSize >= decodedSize);
-    m_decodedSize -= decodedSize;
-
-    // Clearing the ImageSource destroys the extra decoded data used for
-    // determining image properties.
-    decodedSize += m_decodedPropertiesSize;
-    m_decodedPropertiesSize = 0;
-    decodedSizeChanged(-static_cast<long long>(decodedSize));
-}
-
-void ImageSource::didDecodeProperties(unsigned decodedPropertiesSize)
-{
-    if (m_decodedSize)
-        return;
-
-    long long decodedSize = static_cast<long long>(decodedPropertiesSize) - m_decodedPropertiesSize;
-    m_decodedPropertiesSize = decodedPropertiesSize;
-    decodedSizeChanged(decodedSize);
-}
-
-void ImageSource::growFrames()
-{
-    ASSERT(isSizeAvailable());
-    auto newSize = frameCount();
-    if (newSize > m_frames.size())
-        m_frames.grow(newSize);
-}
-
-void ImageSource::setNativeImage(Ref<NativeImage>&& nativeImage)
-{
-    ASSERT(m_frames.size() == 1);
-    ImageFrame& frame = m_frames[0];
-
-    ASSERT(!isDecoderAvailable());
-
-    frame.m_nativeImage = WTFMove(nativeImage);
-
-    frame.m_decodingStatus = DecodingStatus::Complete;
-    frame.m_size = frame.m_nativeImage->size();
-    frame.m_hasAlpha = frame.m_nativeImage->hasAlpha();
-}
-
-void ImageSource::cacheMetadataAtIndex(size_t index, SubsamplingLevel subsamplingLevel)
-{
-    ASSERT(index < m_frames.size());
-    ImageFrame& frame = m_frames[index];
-
-    ASSERT(isDecoderAvailable());
-    if (frame.hasMetadata())
-        return;
-
-    frame.m_subsamplingLevel = subsamplingLevel;
-
-    if (frame.m_decodingOptions.hasSizeForDrawing()) {
-        ASSERT(frame.hasNativeImage());
-        frame.m_size = frame.nativeImage()->size();
-    } else
-        frame.m_size = m_decoder->frameSizeAtIndex(index, subsamplingLevel);
-
-    auto metadata = m_decoder->frameMetadataAtIndex(index);
-    frame.m_orientation = metadata.orientation;
-    frame.m_densityCorrectedSize = metadata.densityCorrectedSize;
-    frame.m_hasAlpha = m_decoder->frameHasAlphaAtIndex(index);
-
-    if (repetitionCount())
-        frame.m_duration = m_decoder->frameDurationAtIndex(index);
-}
-
-void ImageSource::cachePlatformImageAtIndex(PlatformImagePtr&& platformImage, size_t index, SubsamplingLevel subsamplingLevel, const DecodingOptions& decodingOptions)
-{
-    ASSERT(index < m_frames.size());
-    ImageFrame& frame = m_frames[index];
-
-    // Clear the current image frame and update the observer with this clearance.
-    decodedSizeDecreased(frame.clear());
-
-    // Do not cache the NativeImage if adding its frameByes to the MemoryCache will cause numerical overflow.
-    size_t frameBytes = size().unclampedArea() * sizeof(uint32_t);
-    if (!isInBounds<unsigned>(frameBytes + decodedSize()))
-        return;
```

Comment:
We no longer do this. Is it not needed? We still store our decoded size as an unsigned.

Reply:
I put it back.